matchmaking/lobby/faction_column.gd

```diff
@@ -21,13 +22,21 @@ func _ready():
 
 
 func add_player(player):
-	var node = PLAYER_INFO_FACTION_COLUMN.instance()
-	node.player = player
-	container.add_child(node)
-	return UPDATE_PLAYER_STATE.PLAYER_ADDED
+	var has_player = false
+	for node in container.get_children():
+		if node is PlayerInfoFactionColumn and node.player.id == player.id:
+			has_player = true
+			break
+	if not has_player:
+		var node = PLAYER_INFO_FACTION_COLUMN.instance()
+		node.player = player
+		container.add_child(node)
+		return UPDATE_PLAYER_STATE.PLAYER_ADDED
+	else:
```

Comment:
``else`` statement is not useful if your ``if`` has a return statement.

For proper code, I would have inverted the condition, so the ``if`` has just the ``return`` statement currently contained in the ``else``, and the rest of the code has one less indentation level